langserve/server.py

```diff
@@ -85,8 +107,17 @@ def _resolve_model(type_: Union[Type, BaseModel], default_name: str) -> Type[Bas
 
     hash_ = model.schema_json()
 
+    if model.__name__ in _SEEN_NAMES and hash_ not in _MODEL_REGISTRY:
+        # If the model name has been seen before, but the model itself is different
```

Comment:
is the order models get loaded in deterministic? Could this result in an indeterministic openapi spec depending on that order?

Reply:
Depends on order of `add_routes` invocation so I think deterministic in this case. [This is because we're only registering top level model names (which is also why we can't resolve the full issue)]